### DISCUSSION TITLE
PETSc Example Problems in MOOSE

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#29129] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Is it possible in MOOSE to execute the example PETSc programs (i.e. ex35.c) included with the installation of MOOSE? If so are there instructions on how to accomplish this?

**Comment by GiudGiud** [comment]
hello

if you followed a manual install (instead of using conda) certainly. You'll want to navigate to moose/petsc and run examples from there.
If you used conda to install petsc and libmesh I don't know how to do it and I dont think the examples are included in your install

Guillaume

**Reply by TLWise** [reply]
I see that the files are listed in the PETSc folder. I followed the instructions for installation that was provided for Linux operating systems. I did not have to do much during the install except for some copying and pasting. I would expect that my install does not count as a manual install. 

Since I do see the example files listed, would you have an example that I might be able to test that my installed version does on does not work?

**Reply by GiudGiud** [reply]
did you run `conda install moose-dev` or `update_and_rebuild...` ?

**Reply by TLWise** [reply]
The second one.

**Reply by TLWise** [reply]
On second thought. I had to do a clean install and used the install moose-dev.

**Reply by GiudGiud** [reply, marked as answer]
> used the install moose-dev.

then on your system you have a dynamic library of PETSc, you do not have a compiled executable that you could use to run tests afaik.

**Reply by TLWise** [reply]
Got it. Thank you. I did have some success tonight using the multigrid methods included in PETSc with my model of pressure driven flow between fixed plates. The behavior of the solver appears to confirm that multigrid method is being applied.
